roles/lib_openshift/src/test/unit/test_oc_serviceaccount_secret.py

```diff
@@ -1,414 +1,199 @@
 '''
  Unit tests for oc secret add
 '''
-
+import copy
+import json
 import os
-import six
 import sys
-import unittest
 import mock
 
+import pytest
+import six
+
+
 # Removing invalid variable names for tests so that I can
 # keep them brief
 # pylint: disable=invalid-name,no-name-in-module
 # Disable import-error b/c our libraries aren't loaded in jenkins
 # pylint: disable=import-error,wrong-import-position
 # place class in our python path
-module_path = os.path.join('/'.join(os.path.realpath(__file__).split('/')[:-4]), 'library')  # noqa: E501
-sys.path.insert(0, module_path)
-from oc_serviceaccount_secret import OCServiceAccountSecret, locate_oc_binary  # noqa: E402
-
-try:
-    import ruamel.yaml as yaml  # noqa: EF401
-    YAML_TYPE = 'ruamel'
-except ImportError:
-    YAML_TYPE = 'pyyaml'
-
-
-class OCServiceAccountSecretTest(unittest.TestCase):
-    '''
-     Test class for OCServiceAccountSecret
-    '''
-
-    @mock.patch('oc_serviceaccount_secret.locate_oc_binary')
-    @mock.patch('oc_serviceaccount_secret.Utils.create_tmpfile_copy')
-    @mock.patch('oc_serviceaccount_secret.Yedit._write')
-    @mock.patch('oc_serviceaccount_secret.OCServiceAccountSecret._run')
-    def test_adding_a_secret_to_a_serviceaccount(self, mock_cmd, mock_write, mock_tmpfile_copy, mock_oc_binary):
-        ''' Testing adding a secret to a service account '''
-
-        # Arrange
-
-        # run_ansible input parameters
-        params = {
-            'state': 'present',
+MODULE_PATH = os.path.realpath(os.path.join(os.path.dirname(__file__), os.pardir,
+                                            os.pardir, os.pardir, 'library'))
+sys.path.insert(1, MODULE_PATH)
```

Comment:
We need to spread the knowledge in the team to review this during code reviews, avoid using `'/'` etc. :+1: